src/client/app/actions/lineReadings.js

```diff
@@ -26,22 +27,28 @@ function shouldFetchLineReadings(state, meterID, timeInterval) {
 	return readingsForTimeInterval === undefined && !readingsForTimeInterval.isFetching;
 }
 
-function requestLineReadings(meterIDs, timeInterval) {
-	return { type: REQUEST_LINE_READINGS, meterIDs, timeInterval };
+function requestLineReadings(dsIDs, timeInterval, dstype) {
+	return { type: REQUEST_LINE_READINGS, dsIDs, timeInterval, dstype };
 }
 
-function receiveLineReadings(meterIDs, timeInterval, readings) {
-	return { type: RECEIVE_LINE_READINGS, meterIDs, timeInterval, readings };
+function receiveLineReadings(dsIDs, timeInterval, readings, dstype) {
+	return { type: RECEIVE_LINE_READINGS, dsIDs, timeInterval, readings, dstype };
 }
 
-function fetchLineReadings(meterIDs, timeInterval) {
+function fetchLineReadings(dsIDs, timeInterval, type) {
 	return dispatch => {
-		dispatch(requestLineReadings(meterIDs, timeInterval));
+		dispatch(requestLineReadings(dsIDs, timeInterval, type));
 		// The api expects the meter ids to be a comma-separated list.
-		const stringifiedMeterIDs = meterIDs.join(',');
-		return axios.get(`/api/readings/line/${stringifiedMeterIDs}`, {
+		const stringifiedIDs = dsIDs.join(',');
+		let endpoint;
+		if (type === DATA_TYPE_GROUP) {
+			endpoint = '/api/readings/line/groups';
+		} else if (type === DATA_TYPE_METER) {
```

Comment:
Better to use an else here instead of an else if so that the endpoint is never null. Or better yet a ternary operator.

Reply:
Fixed in 08d9f32